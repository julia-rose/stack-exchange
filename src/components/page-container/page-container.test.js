import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import PageContainer from './index';
import SearchContainer from '../search-container';
import ResultsContainer from '../results-container';

describe('Page Container', () => {
  const component = shallow(
    <PageContainer />
  );

  it('renders and matches snapshot', () => {
    const component = renderer.create(
      <PageContainer />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains a SearchContainer subcomponent', () => {
    expect(component.find(SearchContainer)).toHaveLength(1);
  });

  it('updates the search term when setNewSearchTerm() is called', () => {
    const before = component.state('searchTerm');
    component.instance().setNewSearchTerm('jquery');
    const after = component.state('searchTerm');
    expect(before).toEqual('');
    expect(after).toEqual('jquery');
  });

  it('contains a ResultsContainer subcomponent when a searchTerm exists', () => {
    component.instance().setNewSearchTerm('python');
    expect(component.find(ResultsContainer)).toHaveLength(1);
  });
});