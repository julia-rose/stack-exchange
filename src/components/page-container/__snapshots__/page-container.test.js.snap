// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Container renders and matches snapshot 1`] = `
<div
  className="page-container"
>
  <h2
    className="page-container__intro"
  >
    Confused? Overwhelmed? Utterly bewildered?
  </h2>
  <h1
    className="page-container__title"
  >
    Search Stack Overflow!
  </h1>
  <div
    className="search-container"
  >
    <div
      className="radio-buttons"
    >
      <label
        className="option-phrase"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="radio"
          value="phrase"
        />
        Search by phrase
      </label>
      <label
        className="option-tag"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="radio"
          value="tag"
        />
        Search by tag
      </label>
    </div>
    <div
      className="input-container"
    >
      <input
        className="search-input"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        className="search-button"
        onClick={[Function]}
      >
        Search
      </button>
    </div>
  </div>
</div>
`;
